src/test/regress/flaky_tests.md

```diff
@@ -0,0 +1,227 @@
+ citus_move_shard_placement
+ --------------------------
+```
+
+**Example fix**
+```diff
++permutation ... "s1-shard-move-c1-block-writes" "s4-shard-move-sep-block-writes" ...
++permutation ... "s1-shard-move-c1-block-writes" "s4-shard-move-sep-block-writes"("s1-shard-move-c1-block-writes") ...
+```
+
+### Disk size numbers are not exactly like expected
+
+**Issue**: In some tests we show the disk size of a table, but due to various
+postgres background processes such as vacuuming these sizes can change slightly.
+
+**Fix**: Expect a certain range of disk sizes instead of a specific one.
+
+**Example**
+```diff
+ VACUUM (INDEX_CLEANUP ON, PARALLEL 1) local_vacuum_table;
+ SELECT pg_size_pretty( pg_total_relation_size('local_vacuum_table') );
+  pg_size_pretty
+ ----------------
+- 21 MB
++ 22 MB
+ (1 row)
+```
+
+**Example fix**
+```diff
+-SELECT pg_size_pretty( pg_total_relation_size('local_vacuum_table') );
+- pg_size_pretty
++SELECT CASE WHEN s BETWEEN 20000000 AND 25000000 THEN 22500000 ELSE s END
++FROM pg_total_relation_size('local_vacuum_table') s ;
++    s
+ ---------------------------------------------------------------------
+- 21 MB
++ 22500000
+```
+
+
+## Simple things to try
+
+If it's none of the above, than probably the reason why the test is flaky is not
+immediately obvious. There's a few simple things that sometimes help for various reasons
+
+### Don't run test in parallel with others
+
+Check in the schedule if the test is run in parallel with others. If it is,
+remove it from there and check if its still flaky.
+
+**Example**
+```diff
+ test: multi_partitioning_utils replicated_partitioned_table
+-test: multi_partitioning partitioning_issue_3970
++test: multi_partitioning
++test: partitioning_issue_3970
+ test: drop_partitioned_table
+```
+
+### Multiple connections cause issues
+
+Citus its adaptive executor sometimes opens extra connections to do stuff in
+parallel. This happens especially in CI, because CI machines are sometimes slow.
+To disable parallelism and get consistent you can use `citus.max_adaptive_executor_pool_size`.
+
+**Example**
+```
++SET citus.max_adaptive_executor_pool_size TO 1;
+ ALTER TABLE dist_partitioned_table ADD CONSTRAINT constraint1 UNIQUE (dist_col, partition_col);
++RESET citus.max_adaptive_executor_pool_size TO 1;
+```
+
+IMPORTANT: If this helps, this could very well indicate a bug. Check with
+senior/principal engineers if it's expected that it helps in this case.
+
+
+## What to do if this all doesn't work?
+
+If none of the advice above worked, the first thing to try is read the failing
+test in detail and try to understand how it works. Often, with a bit of thinking
+you can figure out why it's failing in the way that it's failing. If you cannot
+figure it out yourself, it's good to ask senior/principal engineers, maybe they
+can think of the reason. Or maybe they're certain that it's an actual bug.
+
+
+## What to do if output can never be consistent?
```

Comment:
would be super useful to add one example to this section as well. I have seen many people struggling to add a good one (including myself;) ))